src/lsdb/loaders/dataframe/dataframe_catalog_loader.py

```diff
@@ -0,0 +1,160 @@
+from typing import Dict, List, NamedTuple, Tuple
+
+import dask.dataframe as dd
+import hipscat as hc
+import pandas as pd
+from dask import delayed
+from hipscat.catalog.catalog_info import CatalogInfo
+from hipscat.io import FilePointer, get_file_pointer_from_path
+from hipscat.pixel_math import HealpixPixel, generate_histogram
+from hipscat.pixel_math.hipscat_id import compute_hipscat_id, healpix_to_hipscat_id
+
+from lsdb.catalog.catalog import Catalog, DaskDFPixelMap
+from lsdb.io.csv_io import read_csv_file_to_pandas
+
+HealpixInfo = NamedTuple("HealpixInfo", [("num_points", int), ("pixels", List[int])])
+
+
+class DataframeCatalogLoader:
+    """Loads a HiPSCat formatted Catalog from a Pandas Dataframe"""
+
+    HISTOGRAM_ORDER = 10
+    PARTITION_SIZE = "100MB"
+
+    def __init__(self, path: str, threshold: int = 50, **kwargs) -> None:
+        """Initializes a DataframeCatalogLoader
+
+        Args:
+            path (str): Path to a CSV file
+            threshold (int): The maximum number of data points per pixel
+            **kwargs: Arguments to pass to the creation of the catalog info
+        """
+        self._check_path_is_valid(get_file_pointer_from_path(path))
+        self.path = hc.io.get_file_pointer_from_path(path)
+        self.threshold = threshold
+        self.catalog_info = CatalogInfo(**kwargs)
```

Comment:
There are only certain `catalog_type` parameters that are ok here. E.g. you couldn't create an AssociationCatalog with this method. Could we add a check to see if it's a valid catalog type for this use case? "object" and "source" are the only valid options.

Reply:
I've added a [method](https://github.com/astronomy-commons/lsdb/blob/116edca3619f2d17d4f993e2959faf88f74aec47/src/lsdb/loaders/dataframe/dataframe_catalog_loader.py#L38) that wraps the creation of the _catalog_info_ and raises an error in case the type isn't valid.